### PR TITLE
Updated Dockerfiles

### PR DESCRIPTION
**Pull request by externl**
This PR reworks the Ice 3.7 `Dockerfile`s. Namely it does the following:

- Update base image to `ubuntu:22.04`
- Create and use an `ice` user
- Fix the default endpoints to use `0.0.0.0` instead of `localhost`

### CODE REVIEW

File: 3.7/glacier2/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 
-ARG GLACIER2_VERSION=3.7.8
-
+ARG GLACIER2_VERSION=3.7.9
 ARG DEBIAN_FRONTEND=noninteractive
 
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv B6391CB2CFBA643D \
-    && echo "deb http://download.zeroc.com/Ice/3.7/ubuntu16.04 stable main" >> /etc/apt/sources.list.d/ice.list \
-    && apt-get update \
+RUN groupadd -r ice && useradd -r -g ice ice
```
Comment (pepone) [inline 3.7/glacier2/Dockerfile:8]: Is this required? I thought the packages already do that https://github.com/zeroc-ice/ice-debian-packaging/blob/a9e4bde1930bb659f9c50e6372092917428c722e/debian/zeroc-glacier2.glacier2router.service#L17

Comment (externl) [reply 3.7/glacier2/Dockerfile:8]: Yes the ice packages do this but this is a Docker best practice to ensure a constant id is created across rebuilds.

File: 3.7/icegridnode/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+
+RUN apt-get update \
     && apt-get install --no-install-recommends --no-install-suggests -y \
     zeroc-icegrid=${ICEGRID_VERSION}-* \
     && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+RUN sed -i '/^Ice.UseSystemdJournal=1/d' /etc/icegridnode.conf \
+    && sed -i 's/^IceGrid\.Node\.Data=.*/IceGrid\.Node\.Data=\/var\/lib\/ice\/icegrid/' /etc/icegridnode.conf \
+    && sed -i 's/^IceGrid\.Node\.Output=.*/#IceGrid\.Node\.Output=\/var\/lib\/ice\/icegrid/' /etc/icegridnode.conf \
+    && sed -i 's/^Ice.Default.Locator=.*/#Ice.Default.Locator=/' /etc/icegridnode.conf \
```
Comment (pepone) [inline 3.7/icegridnode/Dockerfile:30]: Why is this required?

**Review by pepone — APPROVED** [review]
Looks good, I didn't test the images.

File: 3.7/icegridnode/Dockerfile (hunk 2 above)
Comment (externl) [reply 3.7/icegridnode/Dockerfile:30]: The default that we set in the rpm is relative to the default name `node1`, but you would never have two instances running int he container. 

```
IceGrid.Node.Output=/var/lib/ice/icegrid/node1
```
